src/models/scallopUtils.ts

```diff
@@ -391,25 +392,34 @@ export class ScallopUtils {
     }
 
     if (lackPricesCoinNames.length > 0) {
-      const pythConnection = new SuiPriceServiceConnection(
-        this.isTestnet
-          ? 'https://hermes-beta.pyth.network'
-          : 'https://hermes.pyth.network'
-      );
-      const priceIds = lackPricesCoinNames.map((coinName) =>
-        this._address.get(`core.coins.${coinName}.oracle.pyth.feed`)
-      );
+      const endpoints = PYTH_ENDPOINTS[this.isTestnet ? 'testnet' : 'mainnet'];
       try {
-        const priceFeeds =
-          (await pythConnection.getLatestPriceFeeds(priceIds)) || [];
-        for (const [index, feed] of priceFeeds.entries()) {
-          const data = parseDataFromPythPriceFeed(feed, this._address);
-          const coinName = lackPricesCoinNames[index];
-          this._priceMap.set(coinName, {
-            price: data.price,
-            publishTime: data.publishTime,
-          });
-          coinPrices[coinName] = data.price;
+        for (const endpoint of endpoints) {
+          try {
+            const pythConnection = new SuiPriceServiceConnection(endpoint);
+            const priceIds = lackPricesCoinNames.map((coinName) =>
+              this._address.get(`core.coins.${coinName}.oracle.pyth.feed`)
+            );
+            const priceFeeds =
+              (await pythConnection.getLatestPriceFeeds(priceIds)) || [];
+            for (const [index, feed] of priceFeeds.entries()) {
+              const data = parseDataFromPythPriceFeed(feed, this._address);
+              const coinName = lackPricesCoinNames[index];
+              this._priceMap.set(coinName, {
+                price: data.price,
+                publishTime: data.publishTime,
+              });
+              coinPrices[coinName] = data.price;
+            }
+
+            break;
```

Comment:
- Iterate through `endpoints` list, break from the iteration if operation success, else try next endpoint